data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetSszState.java

```diff
@@ -0,0 +1,103 @@
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.io.ByteArrayInputStream;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.tuweni.bytes.Bytes;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.teku.api.ChainDataProvider;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.provider.JsonProvider;
+
+public class GetSszState implements Handler {
+  public static final String ROUTE = "/teku/v1/beacon/states/:state_id";
+  private final ChainDataProvider chainDataProvider;
+  private final JsonProvider jsonProvider;
+
+  public GetSszState(final DataProvider dataProvider, final JsonProvider jsonProvider) {
+    this(dataProvider.getChainDataProvider(), jsonProvider);
+  }
+
+  public GetSszState(final ChainDataProvider chainDataProvider, final JsonProvider jsonProvider) {
+    this.jsonProvider = jsonProvider;
+    this.chainDataProvider = chainDataProvider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get SSZ State",
+      tags = {TAG_TEKU},
+      description = "Download the state SSZ object for given state_id.",
+      pathParams = {@OpenApiParam(name = PARAM_STATE_ID, description = PARAM_STATE_ID_DESCRIPTION)},
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(type = "application/octet-stream")),
+        @OpenApiResponse(status = RES_BAD_REQUEST),
+        @OpenApiResponse(status = RES_NOT_FOUND),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR),
+        @OpenApiResponse(status = RES_SERVICE_UNAVAILABLE, description = SERVICE_UNAVAILABLE)
+      })
+  @Override
+  public void handle(@NotNull final Context ctx) throws Exception {
+    final Map<String, String> pathParamMap = ctx.pathParamMap();
+    ctx.header(Header.CACHE_CONTROL, CACHE_NONE);
+
+    SafeFuture<Optional<Bytes>> future =
+        chainDataProvider.getBeaconStateSsz(pathParamMap.get(PARAM_STATE_ID));
+    ctx.result(
+        future.thenApplyChecked(
+            result -> {
+              if (result.isEmpty()) {
+                ctx.status(SC_NOT_FOUND);
+                ctx.result(
+                    BadRequest.serialize(
+                        jsonProvider,
+                        SC_NOT_FOUND,
+                        "State not found: " + pathParamMap.get(PARAM_STATE_ID)));
+                return null;
+              }
+              ctx.header(
+                  "Content-Disposition",
+                  "filename=\"" + pathParamMap.get(PARAM_STATE_ID) + ".ssz\"");
```

Comment:
I'm slightly worried about using a user specified value in the header - and we definitely can't do it without escaping.  I'd suggest just using the state root encoded as hex.